### ISSUE TITLE
Pre-Trained DeepSpeech 2 Model

### ISSUE THREAD

**Issue by gioannideImp** [NVIDIA/OpenSeq2Seq#419]
Hello,

I am trying to run the pre-trained model of DeepSpeech 2 that you have provided on your website (https://nvidia.github.io/OpenSeq2Seq/html/speech-recognition/deepspeech2.html#pre-trained-model). It seems I cannot find exact bash commands somewhere on your tutorials on how to transcribe one of my wav files that I want to test on your model. Could you please explain on how to do that? I tried: "python run.py --config_file=example_configs/speech2text/ds2_large_8gpus_mp.py --mode=infer --infer_output_file=ds2_out.txt", where the file "ds2_large_8gpus_mp.py" was downloaded from your website, and additionally I stored all these files including model data, index, meta data and checkoint in "experiments/librispeech". I also defined infer_params which was not defined in the file on the website, however I don't understand where do I specify in the infer_params, the wav file that I want to transcribe.

Thank you

**Comment by gioannideImp** [comment]
I have downloaded the DS2 pre-trained model and saved it in "experiments/librispeech" 

I then ran on CPU nodes:

python run.py --config_file=example_configs/speech2text/ds2_large_8gpus_mp.py --mode=infer --infer_output_file=ds2_out.txt 

and got this error:

NotFoundError (see above for traceback): Restoring from checkpoint failed. This is most likely due to a Variable name or other graph key that is missing from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint.

The contents of the configuration file are:
`import tensorflow as tf
from open_seq2seq.models import Speech2Text
from open_seq2seq.encoders import DeepSpeech2Encoder
from open_seq2seq.decoders import FullyConnectedCTCDecoder
from open_seq2seq.data import Speech2TextDataLayer
from open_seq2seq.losses import CTCLoss
from open_seq2seq.optimizers.lr_policies import poly_decay


base_model = Speech2Text
#data_root = "[REPLACE THIS TO THE PATH WITH YOUR LIBRISPEECH DATA]"
data_root =  "/raid/data"

base_params = {
  "random_seed": 0,
  "use_horovod": False,
  "num_gpus": 1,
  "batch_size_per_gpu": 16,
  # "max_steps": 1000,
  "num_epochs": 50,

  "save_summaries_steps": 100,
  "print_loss_steps": 100,
  "print_samples_steps": 5000,
  "eval_steps": 5000,
  "save_checkpoint_steps": 10000,
  "logdir": "experiments/librispeech",

  "optimizer": "Momentum",
  "optimizer_params": {
    "momentum": 0.90,
  },
  "lr_policy": poly_decay,
  "lr_policy_params": {
    "learning_rate": 0.001,
    "power": 0.5,
  },
  "larc_params": {
    "larc_eta": 0.001,
  },
  "dtype": tf.float16,
  # weight decay
  "regularizer": tf.contrib.layers.l2_regularizer,
  "regularizer_params": {
    'scale': 0.0005
  },
  "initializer": tf.contrib.layers.xavier_initializer,

  "summaries": ['learning_rate', 'variables', 'gradients', 'larc_summaries',
                'variable_norm', 'gradient_norm', 'global_gradient_norm'],

  "encoder": DeepSpeech2Encoder,
  "encoder_params": {
    "conv_layers": [
      {
        "kernel_size": [11, 41], "stride": [2, 2],
        "num_channels": 32, "padding": "SAME"
      },
      {
        "kernel_size": [11, 21], "stride": [1, 2],
        "num_channels": 32, "padding": "SAME"
      },
    ],
    "num_rnn_layers": 5,
    "rnn_cell_dim": 800,

    "use_cudnn_rnn": False,
    "rnn_type": "cudnn_gru",
    "rnn_unidirectional": False,

    "row_conv": False,

    "n_hidden": 1600,

    "dropout_keep_prob": 0.5,
    "activation_fn": tf.nn.relu,
    #"data_format": "BCFT", # "channels_first",'BCTF', 'BTFC', 'BCFT', 'BFTC'
  },

  "decoder": FullyConnectedCTCDecoder,
  "decoder_params": {
    "use_language_model": False,

    # params for decoding the sequence with language model
    "beam_width": 512,
    "alpha": 2.0,
    "beta": 1.0,

    "decoder_library_path": "ctc_decoder_with_lm/libctc_decoder_with_kenlm.so",
    "lm_path": "language_model/4-gram.binary",
    "trie_path": "language_model/trie.binary",
    "alphabet_config_path": "open_seq2seq/test_utils/toy_speech_data/vocab.txt",
  },
  "loss": CTCLoss,
  "loss_params": {},
}

train_params = {
  "data_layer": Speech2TextDataLayer,
  "data_layer_params": {
    "num_audio_features": 160,
    "input_type": "spectrogram",
    "augmentation": {'time_stretch_ratio': 0.05,
                     'noise_level_min': -90,
                     'noise_level_max': -60},
    "vocab_file": "open_seq2seq/test_utils/toy_speech_data/vocab.txt",
    "dataset_files": [
      data_root + "/librispeech/librivox-train-clean-100.csv",
      data_root + "/librispeech/librivox-train-clean-360.csv",
      data_root + "/librispeech/librivox-train-other-500.csv",
    ],
    "max_duration": 16.7,
    "shuffle": True,
  },
}

eval_params = {
  "data_layer": Speech2TextDataLayer,
  "data_layer_params": {
    "num_audio_features": 160,
    "input_type": "spectrogram",
    "vocab_file": "open_seq2seq/test_utils/toy_speech_data/vocab.txt",
    "dataset_files": [
      data_root + "/librispeech/librivox-dev-clean.csv",
    ],
    "shuffle": False,
  },
}

infer_params = {
    "data_layer": Speech2TextDataLayer,
    "data_layer_params": {
        "num_audio_features": 64,
        "input_type": "logfbank",
        "vocab_file": "open_seq2seq/test_utils/toy_speech_data/vocab.txt",
        "dataset_files": [
            "data/librispeech/sample.csv", #this is in the form of wav_path,filesize,transcript where the last two are set to UNUSED
        ],
        #"shuffle": False,
    },
}`


Any help to be able to transcribe a WAV file would be appreciated

**Comment by borisgin** [comment]
Can you attach the complete log file and original wav, please?

**Comment by gioannideImp** [comment]
Hello, I have managed to transcribe speech to text using Wave2Letter+, using the same wav file that did not work on DeepSpeech 2. Do you still want me to upload it? It seems it is an error with the checkpoint specifically with the Deepspeech 2 pre-trained model.

**Comment by borisgin** [comment]
No. We will check DS2 checkpoint.

**Comment by youssefavx** [comment]
> I have downloaded the DS2 pre-trained model and saved it in "experiments/librispeech"
> 
> I then ran on CPU nodes:
> 
> python run.py --config_file=example_configs/speech2text/ds2_large_8gpus_mp.py --mode=infer --infer_output_file=ds2_out.txt
> 
> and got this error:
> 
> NotFoundError (see above for traceback): Restoring from checkpoint failed. This is most likely due to a Variable name or other graph key that is missing from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint.
> 
> The contents of the configuration file for infering parameters are:
> `infer_params = { "data_layer": Speech2TextDataLayer, "data_layer_params": { "num_audio_features": 64, "input_type": "logfbank", "vocab_file": "open_seq2seq/test_utils/toy_speech_data/vocab.txt", "dataset_files": [ "data/librispeech/sample.csv", #this is in the form of wav_path,filesize,transcript where the last two are set to UNUSED ], }, }`
> 
> Any help to be able to transcribe a WAV file would be appreciated

Can I ask how you were able to give it the input wav file? Is there a command for that?
